api/src/main/java/com/redhat/ipaas/api/v1/rest/IntegrationPatternDAO.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.api.v1.rest;
+
+import com.redhat.ipaas.api.v1.model.IntegrationPattern;
+import com.redhat.ipaas.api.v1.model.ListResult;
+import com.redhat.ipaas.api.v1.model.WithId;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+
+@Named(IntegrationPattern.KIND)
+@ApplicationScoped
+public class IntegrationPatternDAO implements DataAccessObject<IntegrationPattern> {
+
+    private static final Map<String, IntegrationPattern> PATTERNS = new HashMap<>();
+
+    private static void register(IntegrationPattern pattern) {
+        PATTERNS.put(pattern.getId().orElseThrow(()->new IllegalStateException("Integration Pattern requires an ID.")), pattern);
+    }
+
+    public IntegrationPatternDAO() {
+        register(IntegrationPattern.CHOICE);
+        register(IntegrationPattern.OTHERWISE);
+        register(IntegrationPattern.SPLITTER);
+        register(IntegrationPattern.THROTTLER);
+        register(IntegrationPattern.MESSAGE_FILTER);
+        register(IntegrationPattern.SET_BODY);
+        register(IntegrationPattern.SET_HEDER);
+    }
+
+    @Override
+    public Class<IntegrationPattern> getType() {
+        return IntegrationPattern.class;
+    }
+
+    @Override
+    public IntegrationPattern fetch(String id) {
+        return PATTERNS.get(id);
+    }
+
+    @Override
+    public ListResult<IntegrationPattern> fetchAll() {
+        return new ListResult.Builder<IntegrationPattern>()
+            .items(PATTERNS.values())
+            .totalCount(PATTERNS.size())
+            .build();
+    }
+
+    @Override
+    public IntegrationPattern create(IntegrationPattern entity) {
```

Comment:
Shame we have to add this all in here, we can probably tidy up interfaces here somehow, but anyway at least we've done this in the entrypoint to the REST API so these should hopefully never be called.